samples/subsys/zbus/benchmark/src/benchmark.c

```diff
@@ -0,0 +1,216 @@
+	{                                                                                          \
+		struct external_data_msg *actual_message_data;                                     \
+		struct zbus_channel *chan;                                                         \
+		struct bm_msg msg_received = {0};                                                  \
+                                                                                                   \
+		while (!zbus_sub_wait(&name, &chan, K_FOREVER)) {                                  \
+			zbus_chan_claim(chan, K_NO_WAIT);                                          \
+                                                                                                   \
+			actual_message_data = zbus_chan_msg(chan);                                 \
+			__ASSERT_NO_MSG(actual_message_data->reference != NULL);                   \
+                                                                                                   \
+			memcpy(&msg_received, actual_message_data->reference,                      \
+			       sizeof(struct bm_msg));                                             \
+                                                                                                   \
+			zbus_chan_finish(chan);                                                    \
+                                                                                                   \
+			count += BM_MESSAGE_SIZE;                                                  \
+		}                                                                                  \
+	}                                                                                          \
+                                                                                                   \
+	K_THREAD_DEFINE(name##_id, BM_MESSAGE_SIZE + 196, name##_task, NULL, NULL, NULL, 3, 0, 0);
+
+S_TASK(s1)
+#if (BM_ONE_TO >= 2LLU)
+S_TASK(s2)
+#if (BM_ONE_TO > 2LLU)
+S_TASK(s3)
+S_TASK(s4)
+#if (BM_ONE_TO > 4LLU)
+S_TASK(s5)
+S_TASK(s6)
+S_TASK(s7)
+S_TASK(s8)
+#if (BM_ONE_TO > 8LLU)
+S_TASK(s9)
+S_TASK(s10)
+S_TASK(s11)
+S_TASK(s12)
+S_TASK(s13)
+S_TASK(s14)
+S_TASK(s15)
+S_TASK(s16)
+#endif
+#endif
+#endif
+#endif
+
+#else /* SYNC */
+
+void s_cb(struct zbus_channel *chan);
+
+ZBUS_LISTENER_DEFINE(s1, s_cb);
+
+#if (BM_ONE_TO >= 2LLU)
+ZBUS_LISTENER_DEFINE(s2, s_cb);
+#if (BM_ONE_TO > 2LLU)
+ZBUS_LISTENER_DEFINE(s3, s_cb);
+ZBUS_LISTENER_DEFINE(s4, s_cb);
+#if (BM_ONE_TO > 4LLU)
+ZBUS_LISTENER_DEFINE(s5, s_cb);
+ZBUS_LISTENER_DEFINE(s6, s_cb);
+ZBUS_LISTENER_DEFINE(s7, s_cb);
+ZBUS_LISTENER_DEFINE(s8, s_cb);
+#if (BM_ONE_TO > 8LLU)
+ZBUS_LISTENER_DEFINE(s9, s_cb);
+ZBUS_LISTENER_DEFINE(s10, s_cb);
+ZBUS_LISTENER_DEFINE(s11, s_cb);
+ZBUS_LISTENER_DEFINE(s12, s_cb);
+ZBUS_LISTENER_DEFINE(s13, s_cb);
+ZBUS_LISTENER_DEFINE(s14, s_cb);
+ZBUS_LISTENER_DEFINE(s15, s_cb);
+ZBUS_LISTENER_DEFINE(s16, s_cb);
+#endif
+#endif
+#endif
+#endif
+
+struct bm_msg msg_received = {0};
```

Comment:
drop the (which in this case is double redundant, globals are all zeroed at startup), also `static` (with most functions here as well I guess, see my comment on the first sample)